packages/ckeditor5-table/src/tableediting.js

```diff
@@ -113,9 +112,11 @@ export default class TableEditing extends Plugin {
 		conversion.attributeToAttribute( { model: 'colspan', view: 'colspan' } );
 		conversion.attributeToAttribute( { model: 'rowspan', view: 'rowspan' } );
 
-		// Table heading rows and columns conversion.
+		// Table heading columns conversion (change of heading rows requires reconversion of the whole table).
 		conversion.for( 'editingDowncast' ).add( downcastTableHeadingColumnsChange() );
-		conversion.for( 'editingDowncast' ).add( downcastTableHeadingRowsChange() );
+
+		// Table heading rows change requires reconversion of the whole table.
+		this.listenTo( model, 'applyOperation', headingRowsAttributeChangeHandler( model ) );
```

Comment:
@scofalik & @Reinmar - I need to double check this as this changes how table is converted.

After numerous bugs with changing headings + removing rows we concluded that enough is enough. The table models [sucks](https://github.com/ckeditor/ckeditor5/issues/6506). Downcasting `headingRows` change requires moving `<tr>` in the view from `<thead>` to `<tbody>` or vice-versa. It can fail in multiple scenarios ([#6502](https://github.com/ckeditor/ckeditor5/issues/6502), [#6437](https://github.com/ckeditor/ckeditor5/issues/6437), [#6544](https://github.com/ckeditor/ckeditor5/issues/6544), [#6502](https://github.com/ckeditor/ckeditor5/issues/6502), [#6391](https://github.com/ckeditor/ckeditor5/issues/6391), [#6406](https://github.com/ckeditor/ckeditor5/issues/6406) + #7454) and would potentially never be OK.

Here (#7454) there was a problem with table layout post-fixer which was called after each `enqueueChange()` - and thus we hit limit of what API is allowing us to do. The solution is to wait with conversion and post-fixing after **all** table model changes are done and that `table` attribute converter _would know_ that also `table` children are changing.

Possible considered solutions were:

1.  A "hack": set heading rows to 0, remove/add rows, set heading rows to proper value - but it lacks an API for not converting stuff at once.
2.  Some API to mark that some changes need to be either converted at once or should not trigger model post-fixers.
3.  Mark `table` to be re-rendered at once.

We choose option `3` using `differ.refreshItem()` - which is called on every `headingRows` attribute operation as It requires to also take external operations into account when dealing with this bug. Also the API for many `TableUtils` get simplified but not passing a `batch` instance (:tada:).

---

In other words the solution is to re-render _whole_ table in the view if `headingRows` attribute was changed by any means.

---

What is needed - some kind of :+1: / :-1: - it might impact collaboration features in some way. Do we need better API for requiring re-converting whole chunk of content?

ps.: @Reinmar this is also some sort of ADR proposal (without sections - but have intro, considered steps and proposed solution).

Reply:
Why this is done on `applyOperation` not in a post-fixer?

As a rule of thumb, we try to avoid doing stuff on `applyOperation` - this is the last resort solution. Getting between multiple operations is a risky business. OTOH, I see that you only mark the table to refresh, so no changes in the model tree are done. Might be okay.

The one difference that quickly comes to mind is that you might have an attribute operation for heading rows and then an operation that reverses it, so in the end, there's no change. Post-fixer would not refresh the table, AFAIU. 

So, once again, if it can be done through post-fixer and there are no counterarguments, I think I'd go with post-fixer. But I don't see real big disadvantages in doing this with `applyOperation` either, other than we simply avoid it because it is a more vulnerable "place", if you know what I mean.